src/v/datalake/backlog_controller.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+#include "base/seastarx.h"
+#include "config/property.h"
+#include "metrics/metrics.h"
+
+#include <seastar/core/gate.hh>
+#include <seastar/core/io_priority_class.hh>
+#include <seastar/core/metrics_registration.hh>
+#include <seastar/core/timer.hh>
+#include <seastar/util/log.hh>
+
+namespace datalake {
+
+class backlog_controller {
+public:
+    using sampling_fn = ss::noncopyable_function<long double()>;
+    backlog_controller(sampling_fn, ss::scheduling_group sg);
+
+    ss::future<> start();
+    ss::future<> stop();
+
+private:
+    void update();
+    void setup_metrics();
+
+    sampling_fn _sampling_f;
+    ss::scheduling_group _scheduling_group;
+
+    config::binding<double> _proportional_coeff;
+    config::binding<uint32_t> _setpoint;
+    std::chrono::steady_clock::duration _sampling_interval;
+    ss::timer<> _sampling_timer;
+    long double _current_sample{0};
```

Comment:
nit: mind adding some docs about these? In particular i think it's helpful to at least understand:
- setpoint is the target value we're trying to maintain
- current sample has the same units as setpoint
- current sample is an output of _sampling_f
- maybe how to think about the coefficient, though maybe that's more useful to put in the cluster config property description?